### PR TITLE
Installable PWA

### PR DESCRIPTION
**Pull request by adrianrudnik**
## Description

This is an example on how to get Kimai on the way as an installable PWA app for desktop and mobile devices as discussed in #1560 

## Types of changes
- [x] Added header directory for sw.js to apache .htaccess. What is the correct Dockerfile right now to execute an `a2enmod headers`?
- [x] Added controller to handle localized manifest.json generations.
- [x] Added manifest.json exclusion to firewall.
- [x] Added registration of service worker to app.js

## Checklist
- [x] I verified that my code applies to the guidelines (`composer kimai:code-check`)
- [ ] I updated the documentation (see [here](https://github.com/kimai/www.kimai.org/tree/master/_documentation))
- [x] I agree that this code is used in Kimai and will be published under the [MIT license](https://github.com/kevinpapst/kimai2/blob/master/LICENSE)

## Remarks

I attached a [pwa report](https://github.com/kevinpapst/kimai2/files/4355367/pwa.pdf) that indicates that these changes should satisfy any requirements to be installable. There are other points that could be improved but are not in scope of this.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
 // Hint: if something doesn't work as expected: yarn upgrade --latest
 
 var Encore = require('@symfony/webpack-encore');
+var WorkboxPlugin = require('workbox-webpack-plugin');
 var webpack = require('webpack');
 
 Encore
     .setOutputPath('public/build/')
-    .setPublicPath('build/')
+    .setPublicPath('/build')
```
Comment (kevinpapst) [inline webpack.config.js]: Besides all other changes, which I will have to look at when there is more time - this one is wrong.
The setting is used like this to allow usage in subdirectory scenarios.
The webpack/encore warning is highly annoying and misleading in this case.

**Comment by kevinpapst** [comment]
Thanks for this quick prototype @adrianrudnik - really appreciated!

I never worked with either PWA or service workers. 
As you said in #1560 adding this is a technical debt and for now it doesn't add a "real benefit", does it? Besides being a great example and starting point.

I just want to start a discussion about pros and cons, which I cannot overlook right now.

File: webpack.config.js (hunk 1 above)
Comment (adrianrudnik) [reply webpack.config.js]: ah, wondered why this was changed. Just a draft PR as an idea how to get there :)

**Comment by muety** [comment]
I'd love to have Kimai as a PWA!

**Comment by samuelT2** [comment]
Having Kimai as a PWA would be awesome. I've already added the app to my homescreen on my iPhone.

> ... it doesn't add a "real benefit", does it? ...

I think it would, eg. I could run the app fullscreen (manifest.json: "display":"standalone") and cached. It would more or less behave like a native app. I just made two projects of mine PWA ready, it took some time to get into service workers, but the result is, in my opinion, really cool.

**Comment by kevinpapst** [comment]
I need support with this topic, as I do not have any experience. If someone wants to rebase this on #2902 and send a new PR which I can test, that would be awesome 🚀

**Pull request closed without merging**